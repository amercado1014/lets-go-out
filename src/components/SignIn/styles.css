.signin-form {
  color: white;
  background-color: #20161C;
  height: 80vh;
  width: 50vw;
  margin: 2% auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.signin-div {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.signin-div input,
.signin-button {
  font-size: 20px;
  color: black;
  margin-bottom: 20px;
}

.signin-button {
  background-color: #afd045;
  cursor: pointer;
}

.signin-button:hover {
  color: white;
}

.signin-button:disabled {
  background-color: lightgray;
  color: black;
}
