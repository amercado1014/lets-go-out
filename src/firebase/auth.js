import { auth } from "./firebase";

// Sign Up
export const signUp = (email, password) =>
  auth.createUserWithEmailAndPassword(email, password);

// Sign In
export const signIn = (email, password) =>
  auth.signInWithEmailAndPassword(email, password);

// Sign out
export const signOut = () => auth.signOut();

// Password Reset
export const passwordReset = (email) =>
  auth.sendPasswordResetEmail(email);

// Password Change
export const passwordChange = (password) =>
  auth.currentUser.updatePassword(password);