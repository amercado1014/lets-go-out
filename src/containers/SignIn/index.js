import React, { Component } from "react";
import "./styles.css";
import { SignUpLink } from "../SignUp";
import { withRouter } from "react-router-dom";
import { auth } from "../../firebase";
import * as routes from "../../constants/routes";

export class SignIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      error: null
    };
  }

  handleChange = event => {
    const { value, name } = event.target;
    this.setState({ [name]: value });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { email, password } = this.state;

    auth
      .signIn(email, password)
      .then(authUser => {
        this.setState({
          email: "",
          password: "",
          error: null
        });
      })
      .then(authUser => this.props.history.push(routes.HOME))
      .catch(error => {
        this.setState({ error: error });
      });
  };

  render() {
    const { email, password, error } = this.state;

    const isInvalid = password === "" || email === "";

    return (
      <form onSubmit={this.handleSubmit}>
        <input
          type="text"
          value={email}
          name="email"
          placeholder="Email Address"
          onChange={this.handleChange}
        />
        <input
          type="password"
          value={password}
          name="password"
          placeholder="Password"
          onChange={this.handleChange}
        />
        <button type="submit" disabled={isInvalid}>
          Sign In
        </button>
        {error && <p>{error.message}</p>}
        <SignUpLink />
      </form>
    );
  }
}

export default withRouter(SignIn);
