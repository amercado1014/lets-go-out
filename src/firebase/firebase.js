import * as firebase from "firebase";

const config = {
  apiKey: "AIzaSyBcdjsk8ijIbqaA5RwkZI-Fga0eg4wN-Eo",
  authDomain: "let-s-order-out.firebaseapp.com",
  databaseURL: "https://let-s-order-out.firebaseio.com",
  projectId: "let-s-order-out",
  storageBucket: "let-s-order-out.appspot.com",
  messagingSenderId: "854964870347"
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const auth = firebase.auth();

export { auth };
