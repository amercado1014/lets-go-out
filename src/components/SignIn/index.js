import React, { Component } from 'react';
import './styles.css';
import { SignUpLink } from '../../containers/SignUp';
import { withRouter } from 'react-router-dom';
import { auth } from '../../firebase';
import * as routes from '../../constants/routes';

export class SignIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      error: null
    };
  }

  handleChange = event => {
    const { value, name } = event.target;
    this.setState({ [name]: value });
  };

  handleSubmit = async event => {
    event.preventDefault();
    const { email, password } = this.state;
    const { history } = this.props;
    try {
      await auth.signIn(email, password);
      history.push(routes.HOME);
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const { email, password, error } = this.state;
    const isInvalid = password === '' || email === '';

    return (
      <form 
        onSubmit={this.handleSubmit}
        className="signin-form">
        <div className="signin-div">
          <input
            type="text"
            value={email}
            name="email"
            placeholder="Email Address"
            onChange={this.handleChange}
          />
          <input
            type="password"
            value={password}
            name="password"
            placeholder="Password"
            onChange={this.handleChange}
          />
          <button type="submit" 
            disabled={isInvalid}
            className="signin-button">
          Sign In
          </button>
          {error && <p>{error}</p>}
          <SignUpLink />
        </div>
      </form>
    );
  }
}

export default withRouter(SignIn);
