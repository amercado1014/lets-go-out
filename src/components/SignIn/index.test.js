import React from 'react';
import { SignIn } from '../SignIn';
import { shallow } from 'enzyme';
import { auth } from '../../firebase';
import * as routes from "../../constants/routes";

describe('SignIn', () => {
  let wrapper;
  const mockHistory = { push: jest.fn() };

  beforeEach(() => {
    wrapper = shallow(<SignIn history={mockHistory}/>);
  });

  it('should math the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default', () => {
    expect(wrapper.state()).toEqual({
      email: '',
      password: '',
      error: null
    });
  });

  it('should update state when handle change is called', () => {
    const mockEvent = { target: { value: 'password', name: 'password' } };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('password')).toEqual('password');
  });

  it('should call signIn on handleSubmit with correct params', () => {
    const mockEvent = { preventDefault: jest.fn() };
    auth.signIn = jest.fn();
    wrapper.setState({
      email: 'jim@gmail.com',
      password: 'password'
    });
    const { email, password } = wrapper.state();
    wrapper.instance().handleSubmit(mockEvent);
    expect(auth.signIn).toHaveBeenCalledWith(email, password);
  });

  it('should call history.push with correct params on handleSubmit', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().handleSubmit(mockEvent);
    expect(mockHistory.push).toHaveBeenCalledWith(routes.HOME);
  });

  it('should set state with error if signIn returns an error', async () => {
    const mockEvent = { preventDefault: jest.fn() };
    auth.signIn = jest.fn().mockImplementation(() =>
      Promise.reject({
        message: 'Error'
      })
    );
    await wrapper.instance().handleSubmit(mockEvent);
    await expect(wrapper.state('error')).toEqual('Error');
  });
});
