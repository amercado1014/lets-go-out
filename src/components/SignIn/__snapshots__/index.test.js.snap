// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should math the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="close-div"
>
          <Link
                    className="close-btn"
                    replace={false}
                    to="/"
          >
                    X
          </Link>
</div>,
        <h1>
          Sign In
</h1>,
        <div
          className="signin-div"
>
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
                    value=""
          />
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <button
                    className="signin-button"
                    disabled={true}
                    type="submit"
          >
                    Sign In
          </button>
          <div
                    className="error-div"
          />
          <Unknown />
</div>,
      ],
      "className": "signin-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="close-btn"
            replace={false}
            to="/"
>
            X
</Link>,
          "className": "close-div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "X",
            "className": "close-btn",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "X",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
/>,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <button
              className="signin-button"
              disabled={true}
              type="submit"
>
              Sign In
</button>,
            <div
              className="error-div"
/>,
            <Unknown />,
          ],
          "className": "signin-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "className": "signin-button",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "error-div",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="close-div"
>
            <Link
                        className="close-btn"
                        replace={false}
                        to="/"
            >
                        X
            </Link>
</div>,
          <h1>
            Sign In
</h1>,
          <div
            className="signin-div"
>
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Email Address"
                        type="text"
                        value=""
            />
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <button
                        className="signin-button"
                        disabled={true}
                        type="submit"
            >
                        Sign In
            </button>
            <div
                        className="error-div"
            />
            <Unknown />
</div>,
        ],
        "className": "signin-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="close-btn"
              replace={false}
              to="/"
>
              X
</Link>,
            "className": "close-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "X",
              "className": "close-btn",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "X",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
/>,
              <input
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <button
                className="signin-button"
                disabled={true}
                type="submit"
>
                Sign In
</button>,
              <div
                className="error-div"
/>,
              <Unknown />,
            ],
            "className": "signin-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email Address",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "className": "signin-button",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "error-div",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
